src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -294,6 +305,28 @@ private void UpdateIssues(IEnumerable<IssueMarker> issueMarkers)
             SnapToNewSnapshot(newSnapshot);
         }
 
+        internal /* for testing */ static Core.IFilterableIssue CreateFilterableIssue(Issue issue, ITextSnapshot textSnapshot)
```

Comment:
Extract to another class (like IssueToFilterableIssueConverter), this class already does too much.
It can also receive a list, instead of one Issue, and this way avoid the Where(x=> x!=null)